### PR TITLE
Include roles/lnls-ans-role-conda and roles/lnls-ans-role-sirius-hlacons.

### PR DESCRIPTION
**Pull request by carneirofc**

### CODE REVIEW

**Comment by carneirofc** [comment]
Cons pydm-opi and cons-common are now installed by the role **lnls-ans-role-sirius-hlacons** which depends on **lnls-ans-role-conda**. This is an initial version. I've tested in a virtual machine, still need to fix the molecule tests of hlacons and add some  meaningful ones.

I couldn't find an official conda plugin, for now I'm using shell. Seems to work just fine.

**Review by fernandohds564 — APPROVED** [review]

File: roles/lnls-ans-role-conda/tasks/setup-Debian.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# Install packages
+- name: Install packages
+  apt:
+    name:
+      - "{{ item.file }}"
+    state: present
+    update_cache: true
+    allow_unauthenticated: true
+  with_items:
+    - "{{ conda_apt }}"
```
Comment (lerwys) [inline roles/lnls-ans-role-conda/tasks/setup-Debian.yml:80]: There is something weird here. ``conda_apt`` variable is the repository name or the package name?

File: roles/lnls-ans-role-conda/tasks/setup-Debian.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+# Ansible will call my handler only once even is just one new repository
+# was added, which is what we want
+- name: Add default repositories
+  apt_repository:
+    repo: "{{ item.baseurl }}"
+    state: "{{ item.state }}"
+    filename: "{{ item.file }}"
+    update_cache: true
+  with_items:
+    - "{{ conda_apt }}"
```
Comment (lerwys) [inline roles/lnls-ans-role-conda/tasks/setup-Debian.yml:61]: There is something weird here. ``conda_apt`` variable is the repository name or the package name?

File: roles/lnls-ans-role-conda/tasks/setup-RedHat.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+---
```
Comment (lerwys) [inline roles/lnls-ans-role-conda/tasks/setup-RedHat.yml:1]: We should remove files not being used.

File: roles/lnls-ans-role-conda/templates/buster-repo-pref.j2 [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Package: *
```
Comment (lerwys) [inline roles/lnls-ans-role-conda/templates/buster-repo-pref.j2:1]: Do we need this templates? I think we can remove them completely.

File: roles/lnls-ans-role-conda/templates/jessie-repo-pref.j2 [hunk 5]
```diff
@@ -0,0 +1,11 @@
+Package: *
```
Comment (lerwys) [inline roles/lnls-ans-role-conda/templates/jessie-repo-pref.j2:1]: Do we need this templates? I think we can remove them completely.

File: roles/lnls-ans-role-conda/templates/stretch-repo-pref.j2 [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Package: *
```
Comment (lerwys) [inline roles/lnls-ans-role-conda/templates/stretch-repo-pref.j2:1]: Do we need this templates? I think we can remove them completely.

File: roles/lnls-ans-role-conda/vars/Debian-buster.yml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+---
```
Comment (lerwys) [inline roles/lnls-ans-role-conda/vars/Debian-buster.yml:1]: If all of the debian files are exactly the same we can just use a single `Debian.yml` file. Ansible will use it for all Debian-based distros

File: roles/lnls-ans-role-conda/vars/Debian-buster.yml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    url: https://repo.anaconda.com/pkgs/misc/gpgkeys/anaconda.asc
+    keyring: /etc/apt/trusted.gpg.d/conda-archive-keyring.gpg
+
+__conda_apt:
+  - name: Debian Buster Conda
+    baseurl: "deb [arch=amd64 signed-by=/etc/apt/trusted.gpg.d/conda-archive-keyring.gpg] https://repo.anaconda.com/pkgs/misc/debrepo/conda stable main"
+    description: "Debian Buster Conda - Repository"
+    file: conda
+    state: present
+
```
Comment (lerwys) [inline roles/lnls-ans-role-conda/vars/Debian-buster.yml:14]: Extra newline at the end.

File: roles/lnls-ans-role-sirius-apps/vars/Debian-jessie.yml [hunk 9]
```diff
@@ -8,10 +8,6 @@ __sirius_apps_packages:
   - "liblapack-dev={{ pkg_version_liblapack_dev }}"
   - "libgsl-dev={{ pkg_version_libgsl_dev }}"
 
-__sirius_apps_pip_packages:
-  - "pyvisa=={{ pkg_version_pyvisa }}"
-  - "pyvisa-py=={{ pkg_version_pyvisa_py }}"
```
Comment (lerwys) [inline roles/lnls-ans-role-sirius-apps/vars/Debian-jessie.yml:13]: Does `pyvisa` and `pyvisa-py` are only used for the controls opi screens? @xresende , @fernandohds564 , @carneirofc ? Is this really safe to remove from the role?

Still, even if we remove this we should keep the as an empty list for documentation, as in: `__sirius_apps_pip_packages: []`

File: roles/lnls-ans-role-sirius-hlacons/README.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
+LNLS Ansible Role Sirius HLA Apps
```
Comment (lerwys) [inline roles/lnls-ans-role-sirius-hlacons/README.md:1]: Wrong title name

File: roles/lnls-ans-role-sirius-hlacons/README.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+=================================
+
+This Ansible role configures some defaults Sirius High Level Applications for Sirius Light Source control machines.
+
+## Requirements
+
+- ansible >= 2.4
+- molecule >= 2.20
```
Comment (lerwys) [inline roles/lnls-ans-role-sirius-hlacons/README.md:9]: I think we need ansible >= 2.6 and molecule >= 3.0.

File: roles/lnls-ans-role-sirius-hlacons/README.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+
+## Example Playbook
+
+```yaml
+---
+- hosts: all
+  tasks:
+  - import_role:
+      name: '{{playbook_dir}}'
```
Comment (lerwys) [inline roles/lnls-ans-role-sirius-hlacons/README.md:21]: Ansible lint complain about the syntax '{{playbook_dir}}'. We should use spaces between {{, as in
"{{ playbook_dir }}

File: roles/lnls-ans-role-sirius-hlacons/README.md [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+
+Tests are performed using Molecule. To run them with python virtualenv, issue:
+
+```bash
+    bash -c "\
+        cd ../ && \
+        virtualenv env --python python3 && \
+        source env/bin/activate && \
+        cd lnls-ans-role-sirius-hla && \
+        pip install molecule docker-py && \
```
Comment (lerwys) [inline roles/lnls-ans-role-sirius-hlacons/README.md:40]: molecule, as of version 3.0, does not install test frameworks as a dependency. We must install them separately with:
pip install molecule testinfra yamllint ansible-lint flake8 docker-py

File: roles/lnls-ans-role-sirius-hlacons/README.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+## Tests
+
+Tests are performed using Molecule. To run them with python virtualenv, issue:
+
+```bash
+    bash -c "\
+        cd ../ && \
+        virtualenv env --python python3 && \
+        source env/bin/activate && \
+        cd lnls-ans-role-sirius-hla && \
```
Comment (lerwys) [inline roles/lnls-ans-role-sirius-hlacons/README.md:39]: wrong role name

File: roles/lnls-ans-role-sirius-hlacons/meta/main.yml [hunk 15]
```diff
@@ -0,0 +1,29 @@
+---
+galaxy_info:
+  author: Claudio Carneiro
+  description: Ansible role to install Sirius HLA CONS Apps packages
+  company: CNPEM - LNLS
+  license: BSD
+  min_ansible_version: 2.4
```
Comment (lerwys) [inline roles/lnls-ans-role-sirius-hlacons/meta/main.yml:7]: I think we need ansible >= 2.6

File: roles/lnls-ans-role-sirius-hlacons/playbook.yml [hunk 16]
```diff
@@ -0,0 +1,6 @@
+---
+- hosts: all
+  become: true
+  tasks:
+  - import_role:
+      name: '{{playbook_dir}}'
```
Comment (lerwys) [inline roles/lnls-ans-role-sirius-hlacons/playbook.yml:6]: We should use spaces between `{{` and `}}`: `name: '{{ playbook_dir }}'`

File: roles/lnls-ans-role-sirius-hlacons/tasks/clone-install-app.yml [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+  git:
+    repo: "{{ item.org_url }}/{{ item.repo_name }}"
+    dest: "{{ item.clone_path }}/{{ item.repo_name }}"
+    version: "{{ item.repo_version | default('master') }}"
+    accept_hostkey: "{{ item.git_accept_hostkey | default(omit) }}"
+    ssh_opts: "{{ item.git_ssh_opts | default(omit) }}"
+    force: "{{ item.force_version | default(false) }}"
+
+- name: "Install app at cons Conda env"
+  shell:
```
Comment (lerwys) [inline roles/lnls-ans-role-sirius-hlacons/tasks/clone-install-app.yml:12]: Could we use `command` module for this? It's generally a safer alternative then `shell`, as explained here: https://docs.ansible.com/ansible/latest/modules/command_module.html

File: tasks-desktops.yml [hunk 18]
```diff
@@ -28,6 +28,8 @@
     name: lnls-ans-role-java
 - import_role:
     name: lnls-ans-role-cs-studio
+- import_role:
+    name: lnls-ans-role-conda
```
Comment (lerwys) [inline tasks-desktops.yml:32]: Should we add hlacons role to this as well?

**Review by lerwys — CHANGES_REQUESTED** [review]
Maybe we could use one of those ansible modules for managing conda?

https://github.com/UDST/ansible-conda
https://github.com/evandam/evandam.conda

I think the second one (evandam.conda) is based on the first and you can just copy the `library` directory into our conda role and you should be able to use it.

File: roles/lnls-ans-role-sirius-apps/vars/Debian-jessie.yml (hunk 9 above)
Comment (fernandohds564) [reply roles/lnls-ans-role-sirius-apps/vars/Debian-jessie.yml:13]: We also use `pyvisa` and `pyvisa-py` in our Linac and transport line currinfo IOCs, @lerwys. I think we cannot remove them from this role, @carneirofc. Sorry for not spotting this on my review and thanks for the note, @lerwys.

**Review by xresende — APPROVED** [review]

**Comment by carneirofc** [comment]
I'm closing this pull request and creating a new branch for each role. I think it's the better approach in order to keep things clean.

**Pull request closed without merging**